app/Filament/Pages/EmailConfiguration.php

```diff
@@ -45,61 +48,64 @@ class EmailConfiguration extends Page
 
     protected static string $view = 'filament.pages.email-configuration';
 
-    protected static bool $shouldRegisterNavigation = false;
-
-    // public function getBreadcrumbs(): array
-    // {
-    //     return [
-    //         $this::getUrl() => 'Email Configuration',
-    //     ];
-    // }
-    //
-    // public static function shouldRegisterNavigation(): bool
-    // {
-    //     /** @var User $user */
-    //     $user = auth()->user();
-    //
-    //     return $user->can(['assistant.access_ai_settings']) || ListConsentAgreements::shouldRegisterNavigation();
-    // }
-    //
-    // public function mount(): void
-    // {
-    //     /** @var User $user */
-    //     $user = auth()->user();
-    //
-    //     abort_unless($user->can(['assistant.access_ai_settings']) || ListConsentAgreements::shouldRegisterNavigation(), 403);
-    //
-    //     /** @var NavigationItem $firstNavItem */
-    //     $firstNavItem = collect($this->getSubNavigation())->first(function (NavigationItem $item) {
-    //         return $item->isVisible();
-    //     });
-    //
-    //     if (is_null($firstNavItem)) {
-    //         abort(403);
-    //     }
-    //
-    //     abort_unless($firstNavItem, Response);
-    //
-    //     redirect($firstNavItem->getUrl());
-    // }
-    //
-    // public function getSubNavigation(): array
-    // {
-    //     $navigationItems = $this->generateNavigationItems(
-    //         [
-    //             ListNotificationSettings::class,
-    //         ]
-    //     );
-    //
-    //     /** @var User $user */
-    //     $user = auth()->user();
-    //
-    //     if ($user->can(['assistant.access_ai_settings'])) {
-    //         $navigationItems = [
-    //             ...$navigationItems,
-    //         ];
-    //     }
-    //
-    //     return $navigationItems;
-    // }
+    public function getBreadcrumbs(): array
+    {
+        return [
+            $this::getUrl() => 'Email Configuration',
+        ];
+    }
+
+    public static function shouldRegisterNavigation(): bool
+    {
+        /** @var User $user */
+        $user = auth()->user();
+
+        return $user->can(['notification_setting.view-any']) || ListNotificationSettings::shouldRegisterNavigation();
```

Comment:
I think it would be better here to cycle through `getSubNavigation` and return `true` if any of them `shouldRegisterNavigation()`. That way this menu item is displayed if they have access to any of the subpages.

Reply:
It looks like `generateNavigationItems` already checks `shouldRegisterNavigation` and only returns items that do. So it should be sufficient to check if `getSubNavigation` has at least one visible item.